commands/cluster-slots.md

```diff
@@ -2,35 +2,41 @@
 Redis instances. The command is suitable to be used by Redis Cluster client
 libraries implementations in order to retrieve (or update when a redirection
 is received) the map associating cluster *hash slots* with actual nodes
-network coordinates (composed of an IP address, a TCP port, and the node ID), so that when
+network coordinates (composed of a preferred endpoint, a TCP port, the node ID, and additional networking information), so that when
 a command is received, it can be sent to what is likely the right instance
-for the keys specified in the command.
+for the keys specified in the command. 
+
+The preferred endpoint is the one the one that clients use to send requests, and will be either an IP Address, a hostname, or NULL. The hostname and IP addresses provide the address that should be used to connect to the node. A NULL value indicates the node has an unknown endpoint and the client should connect to the same endpoint it used to send the `CLUSTER SLOTS` command but with the port for the given node. This unknown endpoint configuration is useful when the Redis nodes are behind a load balancer that Redis doesn't know the endpoint of. Which endpoint is set as preferred is determined by the `cluster-preferred-endpoint-type` config.
+
+Additional network information is provided as a map on the fourth argument for each node. The following metadata is available:
+* ip: When the preferred endpoint is not set to ip.
+* hostname: When a node has an announced hostname but the primary endpoint is set not set hostname.
 
 ## Nested Result Array
 Each nested result is:
 
   - Start slot range
   - End slot range
-  - Master for slot range represented as nested IP/Port/ID array
+  - Master for slot range represented as nested preferred endpoint/port/ID array/network metadata
```

Comment:
The operator precedence between spaces and slashes is a bit unclear here. "IP/Port/ID array" i an array of the three elements IP, Port and ID. Now we add more elements at the end of this array, so "preferred-endpoint/port/ID/network-metadata array" may be better, but I think we can list them in a clearer way, such as a bullet list or in redis-cli output style `1) 2) 3)`.

Maybe we also need a name for this to refer to it in the other occurrences below (something shorter than "preferred-endpoint/port/ID/network-metadata array" if possible).

Reply:
I named it "networking information", since that is how it was originally introduced.

Reply:
Also added information in a list about what the networking information for a node looks like. There is a future example in RESP format as well.